pixl_dcmd/src/pixl_dcmd/main.py

```diff
@@ -96,25 +96,42 @@ def anonymise_dicom(dataset: Dataset) -> Dataset:
     return dataset
 
 
-def merge_tag_schemes(tag_operation_files: list[Path]) -> list[dict]:
+def merge_tag_schemes(
+    tag_operation_files: TagOperations, manufacturer: Optional[str] = None
+) -> list[dict]:
     """Merge multiple tag schemes into a single scheme."""
-    all_tags: dict[tuple, dict] = {}
-
-    for tag_operation_file in tag_operation_files:
-        with tag_operation_file.open() as file:
-            # Load tag operations scheme from YAML.
-            tags = yaml.safe_load(file)
-            if not isinstance(tags, list) or not all(
-                [isinstance(tag, dict) for tag in tags]
-            ):
-                raise ValueError(
-                    "Tag operation file must contain a list of dictionaries"
-                )
-            all_tags.update(_scheme_list_to_dict(tags))
+    base_tags = _load_scheme(tag_operation_files.base)
+    all_tags = _scheme_list_to_dict(base_tags)
+
+    if tag_operation_files.manufacturer_overrides is None:
+        return list(all_tags.values())
+
+    manufacturer_overrides = _load_scheme(tag_operation_files.manufacturer_overrides)
+
+    # Keep only the overrides for the specified manufacturer
+    manufacturer_tags = [
+        tag
+        for override in manufacturer_overrides
+        if override["manufacturer"] == manufacturer
+        for tag in override["tags"]
+    ]
```

Comment:
FYI having manufacturer as the dict key would have made this easier, as we could just look up by dict returning empty list if not found. Fine to keep like this because we may end up needing to alter based on software version or model